### DISCUSSION TITLE
Propagation of Data Intervals with Data Aware scheduling

### DISCUSSION THREAD

**Discussion by sirVir** [apache/airflow#34080] [Q&A]
I could not find documentation on how the Data Intervals work with Data-Aware (dataset-triggered) schedules.

### Problem
I am thinking about building a DAG, which would request data based on two Producers DAGs. If one of Producer fails, I would like to skip execution of the Consumer. But as soon as both Producers refresh, I would like to "pass on" information about the period they have processed to the Consumer.

It would seem like Data Aware scheduling is what I need.

### My observations (Airflow 2.6.3):
From what I have seen after running some experiments, the `Data Interval` carries from a **single producer** to a **single consumer** task, but if I execute a catchup there are some discrepancies. I Imagine they could be related to using the `SequentialExecutor` when running locally, which might make the consumer task start while the other producer is updating the dataset? In such cases the updated `Data interval end` carries to the Consumer task. But I still get the same number of producer and consumer executions.

It gets more complicated when I tried to match **multiple producers** to **single consumer**. It seems that that upper and lower Data Interval bounds of the Consumer task are computed based on upper and lower bounds of its Producers. But again, I have seen some of them with different bounds. Are my expectations correct?

How are the Data Intervals defined when creating manual Runs? What about failing tasks?

It would be nice to get some clarifications on the expected/intended behavior.

Neither [Multiple Datasets (from Data-aware scheduling)](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html#multiple-datasets) nor [Data Interval (from DAG Runs)](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dag-run.html#data-interval) explains that well. It would be nice to capture the intention behind those Data Intervals somewhere.

**Comment by potiuk** [comment]
I think it's known defficiency of Data aware scheduling now. And there will be some proposal of follow-ups soon that are addressing that. Just stay tuned to the devlist.

**Reply by sirVir** [reply]
Ok, thanks for confirming it is not just me missing understanding of this feature. I will indeed stay tuned.

**Comment by potiuk** [comment]
BTW. Likely - after the [Airflow Summit](https://airflowsummit.org/) not before.
